src/UseCases/ApplyForMembership/Moderation/ModerationService.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\MembershipContext\UseCases\ApplyForMembership\Moderation;
+
+use WMDE\Fundraising\MembershipContext\Domain\Model\MembershipApplication;
+use WMDE\Fundraising\MembershipContext\Domain\Model\ModerationIdentifier;
+use WMDE\Fundraising\MembershipContext\Domain\Model\ModerationReason;
+use WMDE\Fundraising\MembershipContext\UseCases\ApplyForMembership\ApplicationValidationResult;
+use WMDE\FunValidators\Validators\TextPolicyValidator;
+
+/**
+ * This class is for checking if a membership application needs moderation.
+ * It will be applied **after** the use case has created the application.
+ *
+ * Moderation reasons can be either membership fees that are too high (but still plausible) or text policy violations in
+ * the email or postal address fields ("bad words", according to a deny- and allow-list).
+ *
+ * For forbidden email addresses, we immediately delete the membership application.
+ */
+class ModerationService {
+
+	private const YEARLY_PAYMENT_MODERATION_THRESHOLD_IN_EURO = 1000;
+
+	private TextPolicyValidator $textPolicyValidator;
+	/**
+	 * @var string[]
+	 */
+	private array $emailAddressBlacklist;
```

Comment:
We could take the opportunity and rename the property to `$emailAddressBlocklist`